### PR TITLE
Refine schema for the manifest CRD

### PR DESCRIPTION
**Pull request by edwardstudy**
As story [discussed](https://www.pivotaltracker.com/story/show/161744494), we refined CRD schema which support "one manifest, multiple ops files".

For example, following cr.spec defines base manifest from `configmap/bosh-manifest` and applies two ops from `configmap/bosh-ops` and `secret/bosh-ops-secret`:

```
spec:
  manifest:
    type: "configmap"
    ref: "bosh-manifest"
  ops:
    - type: "configmap"
      ref: "bosh-ops"
    - type: "secret"
      ref: "bosh-ops-secret"
```

Resolver will get bosh-manifest from **config map, secret or URL**(ops file too) and follow `ops` order to resolve manifest. So `secret/bosh-ops-secret` will override `configmap/bosh-ops` operations if they have same ops paths.

And this PR also add more UTs to test resolver functionality.

Best,

@edwardstudy && @zhangtbj, IBM

### CODE REVIEW

**Comment by edwardstudy** [comment]
Hi, @manno. I rebased and set tracker id to first commit.

File: pkg/bosh/manifest/resolver.go [hunk 1]
```diff
@@ -31,52 +33,86 @@ func NewResolver(client client.Client, interpolator Interpolator) *ResolverImpl
...
+
+// getRefData resolves different manifest reference types and returns manifest data
+func (r *ResolverImpl) getRefData(namespace string, manifestType string, manifestRef string, refKey string) (string, error) {
+	var (
+		refData string
+		ok      bool
+	)
+
+	switch manifestType {
+	case bdc.ConfingMapType:
```
Comment (manno) [inline pkg/bosh/manifest/resolver.go]:
```suggestion
	case bdc.ConfigMapType:
```

Comment (edwardstudy) [reply pkg/bosh/manifest/resolver.go]: Fix typo

File: pkg/kube/apis/boshdeploymentcontroller/v1alpha1/types.go [hunk 2]
```diff
@@ -8,10 +8,28 @@ import (
 // It's used as input for the Kube code generator
 // Run "make generate" after modifying this file
 
+// Valid values for ref types
+const (
+	ConfigMapType string = "configmap"
+	SecretType    string = "secret"
+	UrlType       string = "url"
```
Comment (manno) [inline pkg/kube/apis/boshdeploymentcontroller/v1alpha1/types.go]: `go vet` is still complaining about some minor things.

```suggestion
	UrlType       string = "url"
	URLType     string = "url"
```

File: pkg/kube/apis/boshdeploymentcontroller/v1alpha1/types.go [hunk 3]
```diff
@@ -8,10 +8,28 @@ import (
...
+
 // BOSHDeploymentSpec defines the desired state of BOSHDeployment
 type BOSHDeploymentSpec struct {
-	ManifestRef string `json:"manifest-ref"`
-	OpsRef      string `json:"ops-ref"`
+	Manifest Manifest `json:"manifest"`
+	Ops      []Ops    `json:"ops"`
+}
+
+type Manifest struct {
```
Comment (manno) [inline pkg/kube/apis/boshdeploymentcontroller/v1alpha1/types.go]: `make test` complains about missing comments on Manifest. Personally I don't think they make sense, but our CI will fail. We started discussions to change the CI behavior so we can apply linting and vet warnings only when useful.

File: pkg/kube/apis/boshdeploymentcontroller/v1alpha1/types.go [hunk 4]
```diff
@@ -8,10 +8,28 @@ import (
...
+	Type string `json:"type"`
+	Ref  string `json:"ref"`
+}
+
+// OpsRef defines the ops type and location
```
Comment (manno) [inline pkg/kube/apis/boshdeploymentcontroller/v1alpha1/types.go]: Another go vet complain

**Pull request merged**